engine/evaluator/evaluator.go

```diff
@@ -74,7 +80,7 @@ func (e *Evaluator) evaluateTreatment(key string, bucketingKey string, featureFl
 	ctx.AddDependency("evaluator", e)
 	ctx.AddDependency("ruleBasedSegmentStorage", e.ruleBasedSegmentStorage)
 
-	split := grammar.NewSplit(splitDto, ctx, e.logger)
+	split := grammar.NewSplit(splitDto, ctx, e.logger, grammar.NewRuleBuilder(ctx, e.segmentStorage, e.ruleBasedSegmentStorage, e.featureFlagRules, e.ruleBasedSegmentRules, e.logger))
```

Comment:
what about evaluator receiving the entire ruleBuilder instead of passing the config parameters and creating the ruleBuilder here?